core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceCloneSpec.scala

```diff
@@ -0,0 +1,1398 @@
+            .controlledResourceMetadata(new ControlledResourceMetadata().accessScope(AccessScope.SHARED_ACCESS))
+        )
+      )
+    )
+    val cloneResult = new CloneControlledAzureStorageContainerResult()
+      .jobReport(new JobReport().id(UUID.randomUUID().toString))
+    when(wsmDAO.cloneAzureStorageContainer(any, any, any, any, any, any, any)).thenReturn(cloneResult)
+    val leonardoDAO = mock[LeonardoDAO]
+    when(leonardoDAO.createWDSInstance(any, any, any)).thenAnswer(_ => throw new Exception("WDS didn't start"))
+    val monitorRecordDao = mock[WorkspaceManagerResourceMonitorRecordDao]
+    when(monitorRecordDao.create(any)).thenReturn(Future())
+
+    val service = new MultiCloudWorkspaceService(
+      testContext,
+      wsmDAO,
+      mock[BillingProfileManagerDAO],
+      mock[SamDAO],
+      MultiCloudWorkspaceConfig(MultiCloudWorkspaceManagerConfig("app", 1 seconds, 1 seconds), mock[AzureConfig]),
+      leonardoDAO,
+      "MultiCloudWorkspaceService-test",
+      monitorRecordDao,
+      workspaceRepository,
+      mock[BillingRepository]
+    )
+
+    val result = Await.result(
+      service.cloneAzureWorkspace(sourceWorkspace, billingProfile, workspaceRequest, testContext),
+      Duration.Inf
+    )
+
+    result shouldBe destWorkspace
+    verify(wsmDAO).getCloneWorkspaceResult(any(), ArgumentMatchers.eq(wsmCloneJobId), ArgumentMatchers.eq(testContext))
+    verify(wsmDAO).cloneWorkspace(
+      ArgumentMatchers.eq(sourceWorkspace.workspaceIdAsUUID),
+      any(),
+      ArgumentMatchers.eq(workspaceRequest.name),
+      any(),
+      ArgumentMatchers.eq(workspaceRequest.namespace),
+      ArgumentMatchers.eq(testContext),
+      any()
+    )
+    // verify clone container
+    verify(wsmDAO).cloneAzureStorageContainer(any, any, any, any, any, any, any)
+    verify(leonardoDAO).createWDSInstance(any, any, any)
+    verify(monitorRecordDao).create(any)
+  }
+
+  it should "throw an exception if the billing profile was created before 9/12/2023" in {
+    val service = new MultiCloudWorkspaceService(
+      testContext,
+      mock[WorkspaceManagerDAO],
+      mock[BillingProfileManagerDAO],
+      mock[SamDAO],
+      mock[MultiCloudWorkspaceConfig],
+      mock[LeonardoDAO],
+      "MultiCloudWorkspaceService-test",
+      mock[WorkspaceManagerResourceMonitorRecordDao],
+      mock[WorkspaceRepository],
+      mock[BillingRepository]
+    )
+
+    val result = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(
+        service.cloneAzureWorkspace(
+          defaultWorkspace,
+          new ProfileModel().createdDate("2023-09-11T22:20:48.949Z"),
+          WorkspaceRequest("dest-namespace", "dest-name", Map.empty),
+          testContext
+        ),
+        Duration.Inf
+      )
+    }
+
+    result.errorReport.statusCode.value shouldBe StatusCodes.Forbidden
+  }
+
+  it should "fail if the destination workspace already exists" in {
+    // This is covered in the tests for the workspace repository
```

Comment:
I imagine this was accidentally left behind.

If you wish to leave it for documentation purposes, I would suggest disabling the test or commenting it out-- seeing a test case that passed in the logs but actually did nothing would be very confusing if something related to this functionality actually broke.

Reply:
It wasn't left intentionally.
As I was going through tests, I would sometimes leave some that needed additional verification or something, but all the rest had a `todo` in the comments.
I just missed this one when cleaning them up, because I left out the `todo`.